.drone.star

```diff
@@ -0,0 +1,48 @@
+def clang(arch):
+    return {
+        "kind": "pipeline",
+        "name": "%s-clang" % arch,
+        "steps": [
+            {
+                "name": "test",
+                "image": "fedora",
+                "commands": [
+                    "dnf -y install clang cmake ninja-build lld llvm-devel libcxx-devel diffutils",
+                    "mkdir build && cd build",
+                    'env CC=clang CXX=clang++ CXXFLAGS="-UNDEBUG -stdlib=libc++" LDFLAGS="-fuse-ld=lld" cmake -GNinja -DCMAKE_BUILD_TYPE=Release ..',
+                    "ninja -j16",
+                    "ctest --output-on-failure -j24",
+                ],
+            },
+        ],
+
+    }
+
+def gcc(arch):
+    return {
+            "kind": "pipeline",
+            "name": "%s-gcc" % arch,
+            "steps": [
+                {
+                    "name": "test",
+                    "image": "fedora",
+                    "commands": [
+                        "dnf -y install cmake ninja-build g++ llvm-devel diffutils",
+                        "mkdir build && cd build",
+                        'env CC=gcc CXX=g++ CXXFLAGS="-UNDEBUG" LDFLAGS="-fuse-ld=gold" cmake -GNinja -DCMAKE_BUILD_TYPE=Release ..',
+                        "ninja -j16",
+                        "ctest --output-on-failure -j24",
+                    ],
+                },
+            ],
+
+        }
+
+def main(ctx):
+    return [
+        clang("amd64"),
+        clang("arm64"),
+        gcc("amd64"),
+        gcc("arm64"),
```

Comment:
Looks like we're only testing on AMD and ARM architectures. How easy/hard would it be to add in, say, an x86 targeted build?

Reply:
It wouldn't be very hard to add an x86 build in principle, however I'm fairly sure you can't build f18 on x86 as the compile process for parsing.cc will run out before completing.
I might be wrong about this though and I can certainly try it!

Reply:
I just realised I may have been unclear/confused in this reply: I'm referring to x86-32 above. If you meant x86-64 in general, drone is just using amd64 to refer to the x86-64 architecture, it's not specifically talking about AMD cpus.

Reply:
Thank you for the clarification.  The fact that amd64 doesn't mean AMD cpus is indeed confusing.